File: clients/urls.py
from rest_framework import routers
from django.urls import path
from . import views

router = routers.SimpleRouter()
router.register('', views.ClientViewSet, basename='client')
router.register(
    '(?P<client_pk>[^/.]+)/campaigns',
    views.CampaignViewSet,
    basename='campaign'
)
router.register(
    '(?P<client_pk>[^/.]+)/campaigns/(?P<campaign_pk>[^/.]+)/budgets',
    views.BudgetViewSet,
    basename='budget'
)
router.register(
    '(?P<client_pk>[^/.]+)/platforms',
    views.PlatformViewSet,
    basename='platform'
)
router.register(
    '(?P<client_pk>[^/.]+)/platforms/(?P<platform_pk>[^/.]+)/spend',
    views.SpendROViewSet,
    basename='spend'
)
router.register(
    '(?P<client_pk>[^/.]+)/platforms/(?P<platform_pk>[^/.]+)'
    '/spend-upload-records',
    views.SpendUploadRecordViewSet,
    basename='spend_upload'
)

urlpatterns = [
    path(
        '<int:client_pk>/platforms/<int:platform_pk>/spend/upload/',
        views.UploadSpendView.as_view()
    )
]
urlpatterns += router.urls

File: react_source/js/urls/_URLs.js
import budgets from './budgets'
import campaigns from './campaigns';
import clients from './clients';
import platforms from './platforms';
import spend from './spend';

var urls = []
urls = urls.concat(platforms('/clients/:pk/campaigns/platforms'))
urls = urls.concat(spend('/clients/:pk/campaigns/platforms/:plat/spend'))
urls = urls.concat(budgets('/clients/:pk/campaigns/:camp/budgets'))
urls = urls.concat(campaigns('/clients/:pk/campaigns'))
urls = urls.concat(clients('/clients'))

export default urls